sliceutil/sliceutil.go

```diff
@@ -160,7 +160,9 @@ func FilterString(list []string, fun func(string) bool) []string {
 	return ret
 }
 
-// RemoveString removes any occurrence of a string from a list.
+// RemoveString removes any occurrence of a string from a slice.
+// It mutates the underlaying array and returns a new slice containing
```

Comment:
Hmm, the whole mutation of the underlying is a bit dodgy though, maybe better to clone it... Integrity comes before efficiency? 😅

Reply:
Ah yeah, indeed. The caller will still have the original `len`:
https://play.golang.org/p/DADY4bDN8eb

Reply:
yeah, you are right, better mutate it

Reply:
You could just call the FilterString function now, as the implementation looks similar 😄 
```go
func RemoveString(list []string, s string) ([]string) {
  return FilterString(list, func(v string) bool {
    return s != v
  })
}
```

Reply:
Yes, I thought about it, I event just thought into deleting the entire function

But I do not like it that much the callback approach, it gives me JS nightmares, but it's totally a personal taste